### PR TITLE
Added Pokemon type

### LINKED ISSUE #6: Add the Pokemon Type
Obviously, we need a type to represent a Pokemon in a battle. Keep memory footprint in mind when considering what types to use. Feel free to add any enums that you need to accomplish this.

See: https://github.com/StevensSEC/pokemonbattlelib/wiki/Requirements#pokemon

It's not necessary for the first iteration of this type to be perfect. We can change it later if needed.

### Tasks

- [x] Define the contents of `type Pokemon struct` that are necessary to have in a Pokemon battle, or to be able to obtain external metadata. 
    - For example: the Pokemon's name should not be stored in the struct, but the National Pokedex number should since it can be used to look up the Pokemon's name.
- [x] Implement the `Stringer` interface for `Pokemon`. Example: https://tour.golang.org/methods/17

### PR DESCRIPTION
**Pull request by carpenterd777**
- Added fields to pokemon type
- Move data that can be looked up out into getters
- Implemented the Stringer interface on Pokemon, added Gender type
- Added tests for Pokemon String() function

closes #6

### CODE REVIEW

File: pokemon.go [hunk 1]
```diff
@@ -1,11 +1,110 @@
...
+    } else { // the pokemon has a nickname
+        displayName = p.Nickname
+    }
+    
+    return fmt.Sprintf("%v%v\tLv%d\nHP: %d/%d\n", displayName, 
+        p.Gender, p.Level, p.CurrentHP, p.Stats[0])
+
+}
+
+type Level uint8
```
Comment (dyc3) [inline pokemon.go:91]: I don't think `Level` should have its own type. It's just going to make it harder to work with.

File: pokemon.go [hunk 2]
```diff
@@ -1,11 +1,110 @@
...
+
+func (l Level) IsInValidRange() bool {
+    return l > Level(1) && l <= Level(100)
+}
+
+type BaseStat uint8
+
+type IndividualValue uint8
```
Comment (dyc3) [inline pokemon.go:99]: Same thing for BaseStat, IVs, EVs, Stat. I think it's just going to make it harder to work with in the future

File: pokemon.go [hunk 3]
```diff
@@ -1,11 +1,110 @@
...
+    IVs [6]IndividualValue           // values from 0-31 that represents a Pokemon's 'genetic' potential
+    EVs [6]EffortValue               // values from 0-255 that represents a Pokemon's training in a particular stat
+    Nature Nature                    // represents a Pokemon's disposition and affects stats
+    Stats [6]Stat                    // the actual stats of a Pokemon determined from the above data
+    CurrentHP uint                   // the remaining HP of this Pokemon 
+    HeldItem Item                    // the item a Pokemon is holding
+    Moves [4]Move                    // the moves the Pokemon currenly knows
+    Friendship uint8                 // how close this Pokemon is to its Trainer
+    OriginalTrainerName string       // the name of the first Trainer who caught this Pokemon
+    OriginalTrainerID uint16         // a number associated with the first Trainer who caught this Pokemon
```
Comment (dyc3) [inline pokemon.go:21]: Does this have any affect on the battle?

File: pokemon.go [hunk 4]
```diff
@@ -1,11 +1,110 @@
 package pokemonbattlelib
 
+import "fmt"
+
 type Pokemon struct {
-	// National Pokedex Number
-	NatDex uint16
-	// TODO
+    NatDex uint16                    // National Pokedex Number
+    Nickname string                  // player-given name for the Pokemon
```
Comment (dyc3) [inline pokemon.go:7]: For simplicity, lets forgo nicknames on pokemon. it's something we can add in later if needed.

File: pokemon.go [hunk 5]
```diff
@@ -1,11 +1,110 @@
...
+    */
+} 
+
+func (p *Pokemon) ComputeStats() {
+    /* TODO: Implement function to compute
+       Pokemon's actual stats based on the Pokemon's
+       base stats, IVs, EVs, level, and Nature
+       and set the stats to the computed amounts
+    */
+}
```
Comment (dyc3) [inline pokemon.go:72]: Remove these unimplemented functions. They assume too much about the implementation of what other people are working on

File: gender.go [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    Genderless Gender = iota
+    Female
+    Male
+)
+
+// implement Stringer
+
+func (g Gender) String() string {
+    if (g == Genderless) {
+        return ""
```
Comment (dyc3) [inline gender.go:15]: Line not covered in tests

**Review by dyc3 — CHANGES_REQUESTED** [review]

File: pokemon.go (hunk 3 above)
Comment (carpenterd777) [reply pokemon.go:21]: Pokemon who's OTID does not match their current Trainer's gain more EXP in battle. The name might be able to be omitted though

Comment (dyc3) [reply pokemon.go:21]: Let's omit the name then

**Review by dyc3 — CHANGES_REQUESTED** [review]
Looks like some of the tests are failing

**Comment by carpenterd777** [comment]
yeah, the tests are from `agent_test.go`, which is not on this branch. how should I fix it?

**Comment by dyc3** [comment]
rebase on top of the current main branch? or merge main into this branch, either way should work

**Comment by carpenterd777** [comment]
I've tried rebasing and it told me that my branch was already up to date even after running `git fetch`, so I'll try merging now

**Comment by carpenterd777** [comment]
Rebasing made this pr into a massive fusion of the past prs from today, I think should try merging first next time lmao

**Review by dyc3 — APPROVED** [review]

**Pull request merged**